src/systemcmds/system_time/system_time.cpp

```diff
@@ -0,0 +1,127 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <px4_platform_common/module.h>
+#include <px4_platform_common/time.h>
+
+using namespace time_literals;
+
+static void	usage();
+
+extern "C" {
+	__EXPORT int system_time_main(int argc, char *argv[]);
+}
+
+int
+system_time_main(int argc, char *argv[])
+{
+	if (argc >= 2) {
+		if (!strcmp(argv[1], "get")) {
+			//get system time
+			struct timespec ts = {};
+			px4_clock_gettime(CLOCK_REALTIME, &ts);
+			time_t utc_time_sec = ts.tv_sec + (ts.tv_nsec / 1e9);
+
+			//convert to date time
+			char buf[80];
+			struct tm date_time;
+			localtime_r(&utc_time_sec, &date_time);
+			strftime(buf, sizeof(buf), "%a %Y-%m-%d %H:%M:%S %Z", &date_time);
+
+			//get time since boot
+			hrt_abstime since_boot_sec = hrt_absolute_time() / 1_s;
+
+			PX4_INFO("Unix epoch time: %ld", (long)utc_time_sec);
+			PX4_INFO("System time: %s", buf);
+			PX4_INFO("Uptime (since boot): %" PRIu64 " s", since_boot_sec);
+			return 0;
+		}
+
+		if (!strcmp(argv[1], "set")) {
+			if (argc == 3) {
+				//set system time
+				struct timespec ts = {};
+				ts.tv_sec = (time_t)strtoul(argv[2], NULL, 0);
```

Comment:
```suggestion
				ts.tv_sec = (time_t)strtoul(argv[2], nullptr, 0);
```

![Screenshot from 2020-09-25 10-33-07](https://user-images.githubusercontent.com/84712/94279742-8a22f200-ff1a-11ea-83fd-b9a29daac65b.png)